packages/lib-classifier/src/store/SubjectStore.spec.js

```diff
@@ -39,22 +39,34 @@ describe('Model > SubjectStore', function () {
   describe('Actions > advance', function () {
     describe('with a full queue', function () {
       let rootStore
-      let previousSubject
       let initialSize
 
       before(function () {
         rootStore = setupStores()
       })
 
       beforeEach(function () {
-        previousSubject = rootStore.subjects.resources.values().next().value.id
         initialSize = rootStore.subjects.resources.size
         rootStore.subjects.advance()
       })
 
       it('should reduce the queue size by one', function () {
         expect(rootStore.subjects.resources.size).to.equal(initialSize - 1)
       })
+    })
+
+    describe('setting an active subject', function () {
+      let rootStore
+      let previousSubject
+
+      before(function () {
+        rootStore = setupStores()
+      })
+
+      beforeEach(function () {
+        previousSubject = rootStore.subjects.resources.values().next().value.id
```

Comment:
Calling `.next()` advances an iterator. I wonder if that is causing problems since the `advance()` method also advances the `values()` Iterator?